addons/sale/models/sale.py

```diff
@@ -1265,3 +1262,9 @@ def _onchange_discount(self):
             discount = (new_list_price - price) / new_list_price * 100
             if discount > 0:
                 self.discount = discount
+
+    def _get_default_name_based_on_product(self, product):
+        """ compute a default name for this sale order line, based on its product
+        this method exists so it can be overridden in other modules to change how the default name is computed
+        """
```

Comment:
+    def get_default_name_for_sale_order(self):
+        """ compute a default name to be used on a sale order line referencing this product
+        """
+        name = self.display_name
+        if self.description_sale:
+            name += '\n' + self.description_sale
+        return name